packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -55,15 +55,18 @@ final class AssetTransformer {
     required Logger logger,
   }) async {
 
-    String getTempFilePath(int transformStep) {
+    final Directory tempDirectory = _fileSystem.systemTempDirectory.createTempSync();
+
+    File newTempFile(int transformStep) {
       final String basename = _fileSystem.path.basename(asset.path);
       final String ext = _fileSystem.path.extension(asset.path);
-      return '$basename-transformOutput$transformStep$ext';
+
+      return tempDirectory.childFile('$basename-transformOutput$transformStep$ext');
     }
```

Comment:
Consider an app with `assets/image.png` and `assets/2x/image.png`.

One drawback with this approach is that the temporary file paths can't be mapped back to the original file, which can be useful while troubleshooting asset transformations. Consider this output from verbose logs:

```
[        ] Transforming asset using command '/Users/andrewkolos/Documents/GitHub/flutter/bin/cache/dart-sdk/bin/dart run grayscale_transformer --input=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.9R9Ku0/paKIvd/image.png-transformOutput0.png --output=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.9R9Ku0/paKIvd/image.png-transformOutput1.png'
[        ] Transforming asset using command '/Users/andrewkolos/Documents/GitHub/flutter/bin/cache/dart-sdk/bin/dart run grayscale_transformer --input=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.9R9Ku0/ns1fB4/image.png-transformOutput0.png --output=/var/folders/qh/m_ckdxls0xvfc7ccjj13yjjw010s5h/T/flutter_tools.9R9Ku0/ns1fB4/image.png-transformOutput1.png'
```

It's not possible to tell which of these corresponds to `assets/image.png` and which corresponds to `assets/2x/image.png`.

One alternative, more intuitive, solution would be to include the path of the asset (relative to the project directory) into the temp file path. However, I don't believe anything restricts assets to being within the project directory (IIRC there are known "bugs" that occur with such assets depending on target platform). While such a thing might be worth breaking, I want this fix to be cherry-pickable, so I avoided doing this in this PR.

Reply:
> One alternative, more intuitive, solution would be to include the path of the asset (relative to the project directory) into the temp file path

Would it be possible to have the tool store (in memory) the mapping from source input to temp path, solely for the use of surfacing in these kind of trace logs?

In any case, leaving this potential work for later, in a non-CP change SGTM.